### PR TITLE
[skip-ci] Introduce condition to skip CI for pull requests

### PR DESCRIPTION
**Pull request by vepadulano**
If the tag `[skip-ci]` is present in the PR title, the CI builds for that PR will not be executed.

### CODE REVIEW

**Comment by vepadulano** [comment]
The machinery is a bit weird, because the github-actions comment is produced by the https://github.com/root-project/root/blob/master/.github/workflows/test-result-comment.yml action, but only if the portion of the `root-ci` action that produces the event file is actually run https://github.com/root-project/root/blob/e42d3d71483e80efecc4a045a96ae64a2ad5ff96/.github/workflows/root-ci.yml#L484 . I now need to convince `test-result-comment.yml` not to run at all if the related PR was labeled with the tag

**Comment by vepadulano** [comment]
@dpiparo @hahnjo No amount of changes to the https://github.com/root-project/root/blob/master/.github/workflows/test-result-comment.yml seem to do literally anything on this PR, I even changed the name of the job and that is not reflected, I don't know what to outside merging this PR as it is (removing the debug commits ofc)

**Comment by vepadulano** [comment]
You can see what I mean at https://github.com/root-project/root/actions/workflows/test-result-comment.yml

**Comment by hahnjo** [comment]
> @dpiparo @hahnjo No amount of changes to the https://github.com/root-project/root/blob/master/.github/workflows/test-result-comment.yml seem to do literally anything on this PR, I even changed the name of the job and that is not reflected, I don't know what to outside merging this PR as it is (removing the debug commits ofc)

It *could* be that the definition for `workflow_run` are taken from the target branch. I remember that this is the case for `pull_request_target`; for security reasons you don't want to run arbitrary code with "elevated" privileges. The GitHub documentation is unfortunately very scarce on this, so yes I agree, we should modify to the best of our knowledge and then merge. In the better case, it just works or in the worst case, we'll have to live with an empty comment :shrug:

File: .github/workflows/test-result-comment.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ jobs:
   comment-test-results:
     name: Publish Test Results
 
-    if: github.event.workflow_run.event == 'pull_request'
+    if: github.event.workflow_run.event == 'pull_request' && !(github.event.workflow_run.conclusion == 'skipped')
```
Comment (hahnjo) [inline .github/workflows/test-result-comment.yml:16]:
```suggestion
    if: github.event.workflow_run.event == 'pull_request' && github.event.workflow_run.conclusion != 'skipped'
```
maybe for a bit clearer code

**Review by hahnjo — APPROVED** [review]

**Pull request merged**